pl_bolts/models/rl/common/memory.py

```diff
@@ -329,3 +331,97 @@ def update_priorities(self, batch_indices: List, batch_priorities: List) -> None
         """
         for idx, prio in zip(batch_indices, batch_priorities):
             self.priorities[idx] = prio
+
+
+class SharedReplayBuffer:
+    """
+    This replay buffer is shared across the workers
+    """
+
+    def __init__(self, capacity: int, state_shape, action_shape):
+        """
+        Args:
+            capacity: max number of experiences that will be stored in the buffer
+            state_shape: the shape of states being stored
+            action_shape: the shape of actions being stored
+        """
+        self.count = torch.tensor([0], dtype=torch.int64)
+        self.capacity = capacity
+        self.pos = torch.tensor([0], dtype=torch.int64)
+
+        self.states = torch.zeros((capacity, state_shape), dtype=torch.float32)
+        self.next_states = torch.zeros((capacity, state_shape), dtype=torch.float32)
+        self.actions = torch.zeros((capacity, action_shape), dtype=torch.float32)
+        self.rewards = torch.zeros((capacity,), dtype=torch.float32)
+        self.dones = torch.zeros((capacity,), dtype=torch.bool)
+
+        self.count.share_memory_()
+        self.pos.share_memory_()
+        self.states.share_memory_()
+        self.actions.share_memory_()
+        self.next_states.share_memory_()
+        self.rewards.share_memory_()
+        self.dones.share_memory_()
+
+        self.lock = mp.Lock()
+
+    def __len__(self):
+        return self.count
+
+    def sample(self, batch_size: int):
+        """
+        Randomly samples a number of experiences from the stored buffer
+
+        Args:
+            batch_size: number of experiences to sample
+
+        Returns:
+            a batch of tuple np arrays of state, action, reward, done, next_state
+        """
+        if self.count <= batch_size:
+            with self.lock:
+                nr = self.count[0]
+                return [
+                    self.states[:nr].detach().cpu().numpy(),
+                    self.actions[:nr].detach().cpu().numpy(),
+                    self.rewards[:nr].detach().cpu().numpy(),
+                    self.dones[:nr].detach().cpu().numpy(),
+                    self.next_states[:nr].detach().cpu().numpy()
```

Comment:
why convert the torch tensors to numpy? after sampling from the replay buffer, I think in most cases the data will go to a torch module